### PR TITLE
Add the documentation for the release model

### PR DESCRIPTION
**Pull request by anatoliykmetyuk**

### CODE REVIEW

File: docs/docs/usage/version-numbers.md [hunk 1]
```diff
@@ -1,17 +0,0 @@
----
```
Comment (allanrenucci) [inline docs/docs/usage/version-numbers.md]: We have several links to this page on the dotty website (e.g. release blog posts).

Comment (anatoliykmetyuk) [reply docs/docs/usage/version-numbers.md]: Can we make a redirect from that url to the new one?

File: docs/docs/contributing/release.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
+---
+layout: doc-page
+title: Release Procedure
+---
+
+# Model
+The easiest way to produce a release of a GitHub-based open-source software is to tag the most recent commit on the `master` with the version number at regular intervals of time or once a previously agreed milestone is reached. However, this approach to releasing would rest on the assumption that each commit at the `master` branch can potentially be made into a release. We cannot provide the release-grade quality guarantees for each of the `master` commits, though.
+
+This is why, in Dotty, we are using the above method of naive releasing-by-tagging to mark release candidates (RC’s) and not the stable releases. The stable releases are also marked by a tag, but we have a procedure to assure their quality.
```
Comment (biboudis) [inline docs/docs/contributing/release.md]: revised grammatically

> Consequently, in Dotty, we are using the method above--releasing-by-tagging--to mark release candidates (RC’s) and not the stable releases.

File: docs/docs/contributing/release.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+An RC is promoted to a stable release in one release cycle after its creation. The idea is that this way, we have one release cycle's time to examine the release candidate and find critical issues which cannot be allowed into a stable release.
+
+If such issues are found, their fixes end up on a separate branch dedicated to that release. In one release cycle after the RC creation, the RC, along with all its afterthought fixes, is promoted to a stable release by means of tagging it.
```
Comment (biboudis) [inline docs/docs/contributing/release.md]: afterthought  -> subsequent (cannot be used as adjective)

File: docs/docs/contributing/release.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+# Example
+Say we want to release the 0.14.0 version. The process to do so (at a glance).
```
Comment (biboudis) [inline docs/docs/contributing/release.md]: is the following: or Next we describe the process...

File: docs/docs/contributing/release.md [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+## At the Dotty Repo
+1. Tag the latest `master` commit as `0.14.0-RC1`. This commit is the release candidate for the `0.14.0` version.
+2. Create a branch from that commit called `0.14.x`. This branch is intended to host the afterthought fixes to the RC for the issues that cannot be allowed in the `0.14.0` stable release.
```
Comment (biboudis) [inline docs/docs/contributing/release.md]: afterthought -> subsequent

**Comment by anatoliykmetyuk** [comment]
Thanks for the review @biboudis, I've addressed your points.

**Pull request merged**